aws/sqs/main.tf

```diff
@@ -0,0 +1,35 @@
+locals {
+  # This helps avoid queue names ending in "-" or "-.fifo"
+  given_queue_name = var.queue_name == "" ? "" : "-${var.queue_name}"
+  # All fifo queues must end in .fifo, per AWS rules
+  queue_suffix    = var.is_fifo == true ? ".fifo" : ""
+  full_queue_name = "${var.stack}-${var.env}${local.given_queue_name}${local.queue_suffix}"
+}
+
+resource "aws_sqs_queue" "this" {
+  name                        = local.full_queue_name
+  fifo_queue                  = var.is_fifo
+  content_based_deduplication = var.content_based_deduplication
```

Comment:
Maybe we should enforce that this is `false` for non-fifo queues (with a note in the variable description)?
```
content_based_deduplication  = var.is_fifo  && var.content_based_deduplication
```

Reply:
Ah, I was trying to do something like that, while still allowing for a fifo: true / dedupe: false, and I was having problems doing that. `&&` should do it though.